packages/@aws-cdk/aws-ecs/lib/ec2/ec2-task-definition.ts

```diff
@@ -62,12 +79,42 @@ export class Ec2TaskDefinition extends TaskDefinition implements IEc2TaskDefinit
    * Imports a task definition from the specified task definition ARN.
    */
   public static fromEc2TaskDefinitionArn(scope: Construct, id: string, ec2TaskDefinitionArn: string): IEc2TaskDefinition {
+    return Ec2TaskDefinition.fromEc2TaskDefinitionAttributes(
+      scope, id, { taskDefinitionArn: ec2TaskDefinitionArn },
+    );
+  }
+
+  /**
+   * Imports an existing Ec2 task definition from its attributes
+   */
+  public static fromEc2TaskDefinitionAttributes(
```

Comment:
Can't this whole implementation just defer to `TaskDefinition.fromTaskDefinitionAttributes()` ?

Oh I guess it couldn't because the types are different...

Reply:
I wonder if it might be helpful to define a private class that represents the `Import` class for all 3 types. This would be a file like:

```
aws-ecs-/lib/base/_imported-task-definition.ts
                  ^^^ underscore means we plan to not export this file to users
```

And it could contain a:

```
export class ImportedTaskDefinition extends Resource implements ITaskDefinition, IEc2TaskDefinition, IFargateTaskDefinition {
  // ...
}
```

To cut down on the duplication between these 3 methods.

If this sounds too daunting, I'm okay leaving it as is.

Reply:
Thanks. Will try.

A callout is that, there's a little difference in `fromTaskDefinitionArn`: It does have the return type of interface  `ITaskDefinition` , but internally it's a `TaskDefinitionBase` (which implements `ITaskDefinition`) rather than a `Resource` like what the other two import methods return. Don't see the downside of unifying them into `Resource` for now tho.